cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -266,15 +263,12 @@ class AzureHierarchicalNSEnv : public AzureEnvImpl<AzureHierarchicalNSEnv> {
   bool WithHierarchicalNamespace() const final { return true; }
 };
 
-// Placeholder tests
-// TODO: GH-18014 Remove once a proper test is added
-TEST(AzureFileSystem, InitializeCredentials) {
-  auto default_credential = std::make_shared<Azure::Identity::DefaultAzureCredential>();
-  auto managed_identity_credential =
-      std::make_shared<Azure::Identity::ManagedIdentityCredential>();
-  auto service_principal_credential =
-      std::make_shared<Azure::Identity::ClientSecretCredential>("tenant_id", "client_id",
-                                                                "client_secret");
+TEST(AzureFileSystem, InitialiseFilesystemWithDefaultCredential) {
+  AzureOptions options;
+  options.backend = AzureBackend::kAzurite;  // Irrelevant for this test because it
+                                             // doesn't connect to the server.
+  ARROW_EXPECT_OK(options.ConfigureDefaultCredential("dummy-account-name"));
+  EXPECT_OK_AND_ASSIGN(auto default_credential_fs, AzureFileSystem::Make(options));
```

Comment:
It seems that we can use Azurite with the `DefaultAzureCredential`: https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio%2Cblob-storage#azure-sdks

Can we do an operation (`CreateDir()`?) and check the result to verify whether this filesystem is valid or not?

Reply:
I had a bit of a try but I couldn't get the SSL setup working. I'll give it another go but I don't have great hope for making it work in CI even if I get it working locally.

Reply:
OK. I'll also try it.

Reply:
I tried and understand why you said SSL.
`Azure::Identity::DefaultAzureCredential` uses a Bearer token and Azure SDK for C++ rejects it with http: `Bearer token authentication is not permitted for non TLS protected (https) endpoints.`
If we want to use `DefaultAzureCredential` with Azurite, we need to generate a key and certificate pair and use it.

I looked at how to set it to Azure SDK for C++. It seems that we need to `BlobClientOptions::Transport::Transport`:
* https://github.com/Azure/azure-sdk-for-cpp/blob/e328665588d2998fea198972af90d5108e64a968/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp#L174
* https://github.com/Azure/azure-sdk-for-cpp/blob/e328665588d2998fea198972af90d5108e64a968/sdk/core/azure-core/inc/azure/core/internal/client_options.hpp#L97
* https://github.com/Azure/azure-sdk-for-cpp/blob/e328665588d2998fea198972af90d5108e64a968/sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp#L221

If we set `BlobClientOptions::Transport::Transport`, we need to specify curl based HTTP transport implementation or WinHTTP based HTTP transport implementation. They have different configurations for TLS...

How about using `TestAzureHierarchicalNSFileSystem` to test `DefaultAzureCredential`? If we use the real Azure service, we don't need to custom TLS configuration.

Reply:
It would definitely be possible to test against real blob storage but there will be a significant amount of manual configuration for all the identities to test all the different authentications. 

Then we need to provide details of these identities to `TestAzureHierarchicalNSFileSystem` either they are required always or we need to add a new versions for each auth e.g. `TestAzureHierarchicalNSFileSystemWithServicePrincipal`.

Also there are some Auth methods that are not feasible to test. For example managed identity can only work on Azure VMs and workload identity can only work in kubernetes. 

Personally I don't think this is worthwhile because of how little complexity there is outside the Azure SDK.

Reply:
I think this test is enough, because as @pitrou said the other day: "we are not re-implementing the Azure SDK".

Reply:
> Also there are some Auth methods that are not feasible to test. For example managed identity can only work on Azure VMs and workload identity can only work in kubernetes.

Oh...

OK. I see.